### PR TITLE
feat: checksum_algo param for find module

### PR DESCRIPTION
**Pull request by ShIRannx**
##### SUMMARY

This PR introduces a new checksum_algorithm option to the find module in cases where the get_checksum option is selected. The checksum_algorithm option allows users to specify the algorithm of checksum returned.
##### ISSUE TYPE

- Feature Pull Request

##### ADDITIONAL INFORMATION

```yaml
- name: Find files with md5 checksum_algo return
  ansible.builtin.find:
    paths: /path/to/search
    get_checksum: true
    checksum_algorithm: md5
  register: found_files
```

### CODE REVIEW

File: lib/ansible/modules/find.py [hunk 1]
```diff
@@ -154,6 +154,17 @@
...
+        description:
+        - Algorithm to determine checksum of file.
+        - Will throw an error if the host is unable to use specified algorithm.
+        - The remote host has to support the hashing method specified, V(md5)
+            can be unavailable if the host is FIPS-140 compliant.
+        type: str
+        choices: [ md5, sha1, sha224, sha256, sha384, sha512 ]
+        default: sha1
+        aliases: [ checksum, checksum_algo ]
+        version_added: "2.17"
```
Comment (bcoca) [inline lib/ansible/modules/find.py]: 2.18 at this point

Comment (bcoca) [reply lib/ansible/modules/find.py]: add note, availability might be restricted by the target system, for example FIPS systems won't allow md5 use

**Review by bcoca — CHANGES_REQUESTED** [review]
mostly lgtm, needs minor changes and to pass tests

File: lib/ansible/modules/find.py [hunk 2]
```diff
@@ -154,6 +154,18 @@
             - When doing a C(contains) search, determine the encoding of the files to be searched.
         type: str
         version_added: "2.17"
+    checksum_algorithm:
+        description:
+        - Algorithm to determine checksum of file.
+        - Will throw an error if the host is unable to use specified algorithm.
+        - The remote host has to support the hashing method specified, V(md5)
+            can be unavailable if the host is FIPS-140 compliant.
+        - Availability might be restricted by the target system, for example FIPS systems won't allow md5 use
```
Comment (bcoca) [inline lib/ansible/modules/find.py]: redundant phrases, also update get_checksum description

Comment (bcoca) [reply lib/ansible/modules/find.py]: also might go into a docfragment and share with stat module

Comment (ShIRannx) [reply lib/ansible/modules/find.py]: May I add my name after the Copyright tag in the docfragment file, or should I just keep it as 'Ansible Project'?

Comment (ShIRannx) [reply lib/ansible/modules/find.py]: additionally, checksum_algorithm is added between stat and find modules in different ansible version. how do i present it on docstring?

**Comment by ShIRannx** [comment]
Excuse me, idk the meaning of sanity error `version_added for existing option (get_checksum) should belong to collection 'ansible.builtin'. Currently belongs to None (75%)` and how to handle it. any advise here?

**Comment by ShIRannx** [comment]
Excuse me, is anyone following up on this?

**Comment by konstruktoid** [comment]
Try this:
```diff
diff --git a/lib/ansible/modules/find.py b/lib/ansible/modules/find.py
index dd8dbb5e74..a51fe2eb3b 100644
--- a/lib/ansible/modules/find.py
+++ b/lib/ansible/modules/find.py
@@ -34,6 +34,10 @@ options:
             - Whether to return a checksum of the file.
         default: no
     checksum_algorithm:
+        description:
+            - The checksum algorithm to be retrieved.
+        type: str
+        default: "sha1"
         version_added: "2.18"
     patterns:
         default: []
```

File: lib/ansible/modules/find.py [hunk 3]
```diff
@@ -154,7 +156,7 @@
             - When doing a C(contains) search, determine the encoding of the files to be searched.
         type: str
         version_added: "2.17"
-extends_documentation_fragment: action_common_attributes
+extends_documentation_fragment: [action_common_attributes, checksum_common]
```
Comment (Akasurde) [inline lib/ansible/modules/find.py:159]: I do not see any `checksum_common` doc_fragment.

File: lib/ansible/modules/find.py [hunk 4]
```diff
@@ -29,6 +29,13 @@
             - You can choose seconds, minutes, hours, days, or weeks by specifying the
               first letter of any of those words (e.g., "1w").
         type: str
+    get_checksum:
+        description:
+            - Whether to return a checksum of the file.
+        default: no
+        version_added: "ansible.builtin"
```
Comment (Akasurde) [inline lib/ansible/modules/find.py]:
```suggestion
        version_added: historical
```

File: lib/ansible/modules/stat.py [hunk 5]
```diff
@@ -61,6 +44,7 @@
     version_added: "2.3"
 extends_documentation_fragment:
   -  action_common_attributes
+  - checksum_common
```
Comment (Akasurde) [inline lib/ansible/modules/stat.py]:
```suggestion
  -  action_common_attributes
  -  checksum_common
```

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/find.py [hunk 6]
```diff
@@ -29,6 +29,13 @@
             - You can choose seconds, minutes, hours, days, or weeks by specifying the
               first letter of any of those words (e.g., "1w").
         type: str
+    get_checksum:
+        description:
+            - Whether to return a checksum of the file.
+        default: no
+        version_added: historical
```
Comment (Akasurde) [inline lib/ansible/modules/find.py]:
```suggestion
        version_added: 'ansible.builtin'
```

**Review by Akasurde — CHANGES_REQUESTED** [review]

**Review by Akasurde — APPROVED** [review]
LGTM

**Review by bcoca — APPROVED** [review]

**Pull request merged**